platform/linux-generic/arch/mips64/odp/api/abi/cpu_inlines.h

```diff
@@ -0,0 +1,56 @@
+/* Copyright (c) 2016-2018, Linaro Limited
+ * Copyright (c) 2021, Nokia
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier:     BSD-3-Clause
+ */
+
+#ifndef ODP_ARCH_CPU_INLINES_H_
+#define ODP_ARCH_CPU_INLINES_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdint.h>
+
+static inline void _odp_cpu_pause(void)
+{
+	__asm__ __volatile__ ("nop");
+	__asm__ __volatile__ ("nop");
+	__asm__ __volatile__ ("nop");
+	__asm__ __volatile__ ("nop");
+}
+
+uint64_t _odp_cpu_cycles(void)
+{
+	#define CVMX_TMP_STR(x) CVMX_TMP_STR2(x)
+	#define CVMX_TMP_STR2(x) #x
+	uint64_t cycle;
+
+	__asm__ __volatile__ ("rdhwr %[rt],$" CVMX_TMP_STR(31) :
```

Comment:
CVMX_TMP_STR seems unnecessary. This could be just `"rdhwr %[rt],$31"`. And a comment explaining the magic number.

I suppose $31 is specific to Octeon, since mips64 says that $31 is "reserved for implementation-dependent use", while $2 is cycle counter.

Reply:
In this PR I'm only moving the current implementation as-is (e.g. this function was originally in `platform/linux-generic/arch/mips64/odp_cpu_cycles.c`). This function could be optimized in a separate PR.

However, before we spend more time on this we should decide if we still need to support mips64. Currently, the architecture is not tested in the CI at all.

Reply:
Ok, agree. I got confused for a moment there, because github hides removed files.